core/src/main/java/com/linecorp/armeria/server/docs/MethodInfo.java

```diff
@@ -130,7 +152,6 @@ public MethodInfo(String serviceName, String name,
 
         this.returnTypeSignature = requireNonNull(returnTypeSignature, "returnTypeSignature");
         this.parameters = ImmutableList.copyOf(requireNonNull(parameters, "parameters"));
-        assert !useParameterAsRoot || this.parameters.size() == 1;
```

Comment:
Note: the following is also a valid method that has multiple resolvers, but uses headers/query params instead of the body to fill the bean.

Since the last non-param/query will be used, in the worst case it is possible that wrong autocomplete/recommendations are done.

```java
@Post("/bean2")
public HttpResponse bean2(@RequestObject JsonRequest jsonRequest, CompositeBean compositeBean1) {
    return HttpResponse.of("world");
}
```

Having said this, I'm not sure if there is a non-trivial way to intelligently find the parameter that should be used as the body.
Let me think about this a little more..